src/pages/Login.js

```diff
@@ -43,22 +48,22 @@ class Login extends Component {
       }
     });
 
-    const { username, password } = this.state;
     let success = false;
-    if(formIsValid) {
-      let res = await api.login(username, password);
+    const { username, password } = this.state;
+    if (formIsValid) {
+      const res = await api.login(username, password);
       if (res.code !== api.code.ok) {
         this.setState({ error: res.msg });
-        console.log("---- Error ----");
+        console.log('---- Error ----');
         console.log(res);
       } else {
         success = true;
       }
     }
-    this.setState({ loading: false});
-    if(success){
+    this.setState({ loading: false });
+    if (success) {
       // TODO: set loggin state instead
-      this.props.history.push("/dashboard");
+      this.props.history.push('/dashboard');
       window.location.reload();
```

Comment:
Is it possible to avoid the reload?
Jump to '/dashboard' as default is nice, but we should allow a parameter to override the target page.